native/acter/src/api/room.rs

```diff
@@ -522,6 +524,85 @@ impl Room {
             .await?
     }
 
+    pub async fn add_parent_room(&self, room_id: String, canonical: bool) -> Result<String> {
+        let room_id = OwnedRoomId::try_from(room_id)?;
+        if !self
+            .get_my_membership()
+            .await?
+            .can(crate::MemberPermission::CanLinkSpaces)
+        {
+            bail!("You don't have permissions to add parent-rooms");
+        }
+        if !self.is_joined() {
```

Comment:
This will never happen as the `get_my_membership().await?` above will fail before, no? Better to move this check before (as it is also a fairly quick match)

Reply:
same applies to the other circumstances with the same pattern in this PR.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1089/commits/6dae3937da73c512f15b34f5f5dac5bd9b1254d4